src/realm/index_string.hpp

```diff
@@ -121,11 +121,6 @@ class ClusterColumn {
         , m_column_key(column_key)
     {
     }
-    ClusterColumn(const ClusterColumn& other)
```

Comment:
These were copy constructors that a) didn't have a matching operator=, and b) seem to do the same as the default ones. Using an implicitly defaulted operator= with an explicitly declared copy constructor (or destructor) has been deprecated since C++11 (http://eel.is/c++draft/depr.impldec).

This change make it just use the default copy operations implicitly. It also makes the implicit move operations available. If the latter change is undesired, I can explicitly default the copy operators.